backend/Tests/Aggregates/Contract/Types/ModuleSchemaVersionTests.cs

```diff
@@ -0,0 +1,30 @@
+using Application.Aggregates.Contract.Types;
+using FluentAssertions;
+
+namespace Tests.Aggregates.Contract.Types;
+
+public sealed class ModuleSchemaVersionTests
+{
+    [Theory]
+    [InlineData(ModuleSchemaVersion.Undefined, (byte)0)]
+    [InlineData(ModuleSchemaVersion.V0, (byte)0)]
+    [InlineData(ModuleSchemaVersion.V2, (byte)2)]
+    public void WhenMapModuleSchemaVersionToFFIOption_ThenCorrect(
+        ModuleSchemaVersion version, byte mapped
+        )
+    {
+        // Act
+        var ffiOption = ModuleSchemaVersionExtensions.Into(version);
+
+        // Assert
+        if (ModuleSchemaVersion.Undefined == version)
+        {
+            ffiOption.is_some.Should().Be(0);
+        }
+        else
+        {
+            ffiOption.is_some.Should().Be(1);
+            ffiOption.t.Should().Be(mapped);
+        }
```

Comment:
Not a big problem here, but I think conditions in tests should be avoided. It would be much clearer to me if it was just 3 cases written directly.